src/main/java/com/azure/cosmos/sample/async/AsyncMain.java

```diff
@@ -242,47 +225,41 @@ private void queryItems() {
         //  <QueryItems>
         // Set some common query options
 
+        int preferredPageSize = 10; // We'll use this later
+
         CosmosQueryRequestOptions queryOptions = new CosmosQueryRequestOptions();
-        //  Set query metrics enabled to get metrics around query executions
+
+        //  Set populate query metrics to get metrics around query executions
         queryOptions.setQueryMetricsEnabled(true);
 
         CosmosPagedFlux<Family> pagedFluxResponse = container.queryItems(
-            "SELECT * FROM Family WHERE Family.lastName IN ('Andersen', 'Wakefield', 'Johnson')", queryOptions, Family.class);
-
-        final CountDownLatch completionLatch = new CountDownLatch(1);
-
-        pagedFluxResponse.byPage(10).subscribe(
-            fluxResponse -> {
-                System.out.println("Got a page of query result with " +
-                    fluxResponse.getResults().size() + " items(s)"
-                    + " and request charge of " + fluxResponse.getRequestCharge());
-
-                System.out.println("Item Ids " + fluxResponse
-                    .getResults()
-                    .stream()
-                    .map(Family::getId)
-                    .collect(Collectors.toList()));
-            },
-            err -> {
-                if (err instanceof CosmosException) {
-                    //Client-specific errors
-                    CosmosException cerr = (CosmosException)err;
-                    cerr.printStackTrace();
-                    System.err.println(String.format("Read Item failed with %s\n", cerr));
-                } else {
-                    //General errors
-                    err.printStackTrace();
-                }
-
-                completionLatch.countDown();
-            },
-            () -> {completionLatch.countDown();}
-        );
+                "SELECT * FROM Family WHERE Family.lastName IN ('Andersen', 'Wakefield', 'Johnson')", queryOptions, Family.class);
 
         try {
-            completionLatch.await();             
-        } catch (InterruptedException err) {
-            throw new AssertionError("Unexpected Interruption",err);
+
+            pagedFluxResponse.byPage(preferredPageSize).flatMap(fluxResponse -> {
```

Comment:
please remove the whole block with this:

```suggestion
List<String> ids = pagedFluxResponse.byPage(preferredPageSize).getElements().stream().map(Family::getId)
logger.info("Item Ids {}" , ids);
```